textile/chat-features.textile

```diff
@@ -191,19 +191,41 @@ Broadly speaking, messages are published via REST calls to the Chat HTTP API and
 
 * @(CHA-M1)@ Chat messages for a Room are sent on a corresponding realtime channel @<roomId>::$chat::$chatMessages@. For example, if your room id is @my-room@ then the messages channel will be @my-room::$chat::$chatMessages@.
 * @(CHA-M2)@ A @Message@ corresponds to a single message in a chat room. This is analogous to a single user-specified message on an Ably channel (NOTE: **not** a @ProtocolMessage@).
-** @(CHA-M2a)@ @[Testable]@ A @Message@ is considered before another @Message@ in the global order if the @timeserial@ of the corresponding realtime channel message comes first.
-** @(CHA-M2b)@ @[Testable]@ A @Message@ is considered after another @Message@ in the global order if the @timeserial@ of the corresponding realtime channel message comes second.
-** @(CHA-M2c)@ @[Testable]@ A @Message@ is considered to be equal to another @Message@ if they have the same timeserial.
+<div class=deprecated>
+** @(CHA-M2a)@ @[Testable]@ @[Deprecated]@ A @Message@ is considered before another @Message@ in the global order if the @timeserial@ of the corresponding realtime channel message comes first.
+** @(CHA-M2b)@ @[Testable]@ @[Deprecated]@ A @Message@ is considered after another @Message@ in the global order if the @timeserial@ of the corresponding realtime channel message comes second.
+** @(CHA-M2c)@ @[Testable]@ @[Deprecated]@ A @Message@ is considered to be equal to another @Message@ if they have the same timeserial.
+</div>
+** @(CHA-M2d)@ @[Testable]@ A @Message@ is considered before another @Message@ in the global order if the @serial@ of the corresponding realtime channel message comes first.
+** @(CHA-M2e)@ @[Testable]@ A @Message@ is considered after another @Message@ in the global order if the @serial@ of the corresponding realtime channel message comes second.
+** @(CHA-M2f)@ @[Testable]@ A @Message@ is considered to be equal to another @Message@ if they have the same serial.
```

Comment:
I know the wording is the same as before, but I personally find if hard to get for message sorting. Also message equality isn't really true (it is for sorting position but if you only read one spec point it doesn't say that) if serials are the same because one of the messages is an update for the the other one.

The other thing I think we should use `newer` and `older` as opposed to which comes first or second to be clearer. Which one's first depends on sorting direction.

I had a go to rewrite:

> In global order, messages are to be sorted by their `serial`. If the `serial` of a message is less than that of another message, the first message is older than the first. Similarly, if the `serial` of the first message is greater than that of the second message, then the first message is newer.

> If two realtime messages have the same serial, then they represent the same chat message.

Reply:
Perhaps worth defining less/more as lexicographical string comparisons?

Reply:
How about this

```
** @(CHA-M2d)@  A @Message@ contains a unique, immutable @serial@, which is a lexicographically sortable string. This provides a global ordering for all received messages.
** @(CHA-M2e)@ @[Testable]@ In global ordering, a @Message@ is considered to occur before another @Message@ if the @serial@ of the first @Message@ is before the latter when lexicographically sorted.
** @(CHA-M2f)@ @[Testable]@ In global ordering, a @Message@ is considered after another @Message@ if the @serial@ of the first @Message@ is after the latter when lexicographically sorted.
** @(CHA-M2g)@ @[Testable]@ Two @Messages@ are considered to be the same if they have the same @serial@, that is to say, both @Serial@ strings are identical.
```

Reply:
Or we could use the wording like
`is lexicographically smaller than that of the latter.` rather than `is before the latter when lexicographically sorted.`?

Reply:
I like @splindsay-92's: [this one I put a 👍 to.](https://github.com/ably/specification/pull/216#discussion_r1838211483)